src/core/posix.cc

```diff
@@ -19,11 +19,28 @@
  * Copyright (C) 2014 Cloudius Systems, Ltd.
  */
 
+#ifdef SEASTAR_MODULE
+module;
+#endif
+
+#include <memory>
+#include <cassert>
+#include <set>
+#include <vector>
+#include <functional>
+#include <fmt/format.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/mman.h>
+#include <sys/inotify.h>
+
+#ifdef SEASTAR_MODULE
+module seastar;
```

Comment:
I see some logic in the #includes now. I wonder if we can move it to the header files?

```c++
#define SEASTAR_BEGIN_INCLUDES module;

#define SEASTAR_END_INCLUDES module seastar;
```

?

Reply:
i guess this comment is in the same theme of the previous one: to reduce the `#if .. #endif` s. they are probably distracting. but i want to make the macro names reflect what they are doing instead of hiding the internals with names not quite relevant. as, IMHO,  `module` and `module seastar` is not strictly related to header files. they are here to split the module implementation unit into different fragments. and more importantly, `module seastar;` does not mark the end of `#includes`. on the contrary, it declare a module implementation unit.

so, in comparison to the `SEASTAR_MODULE(x...)` suggestion, i am in favor of the latter.  but again, could you re-evaluate the ugly-ness level ?

Reply:
I want to keep the C++ keywords visible and not hide them. It's tricky enough without the reader having to learn a new wrapper.

Reply:
i am confused. if we go with:
```c++
#define SEASTAR_BEGIN_INCLUDES module;
```
the `module` keyword is hidden by the macro. so with existing approach, this keyword is more visible, IMHO.